core/instance.go

```diff
@@ -462,16 +462,13 @@ func (i *instance) createRunInstancesInput(instanceType string, price float64) *
 		TagSpecifications: i.generateTagsList(),
 	}
 
-	if i.IamInstanceProfile != nil {
-		retval.IamInstanceProfile = &ec2.IamInstanceProfileSpecification{
-			Arn: i.IamInstanceProfile.Arn,
-		}
-	}
-
 	if i.asg.LaunchTemplate != nil {
 		retval.LaunchTemplate = &ec2.LaunchTemplateSpecification{
-			LaunchTemplateId:   i.asg.LaunchTemplate.LaunchTemplateId,
-			LaunchTemplateName: i.asg.LaunchTemplate.LaunchTemplateName,
+			LaunchTemplateId: i.asg.LaunchTemplate.LaunchTemplateId,
```

Comment:
There is a subtle bug here. getting the launch template id from the LaunchTemplateSpecification without specifying the version means you will get the Default version of the launch template, which may be *far* different that the version currently being used.

Reply:
@dpikedpike good catch!

I'm not very familiar with the launch templates yet, can you suggest how we could improve this to get the currently used launch template version?

Feel free to add to patches addressing this issue on top of this PR.

Reply:
Code should be:
		retval.LaunchTemplate = &ec2.LaunchTemplateSpecification{
			LaunchTemplateId: i.asg.LaunchTemplate.LaunchTemplateId,
			Version:          i.asg.LaunchTemplate.Version,
		}

The version then comes from the version of the LT being used by the asg

Reply:
Fixed in my latest commit